code/parse/sexp.cpp

```diff
@@ -23298,6 +23567,706 @@ void sexp_set_motion_debris(int node)
+				container.container_name.c_str(),
+				key.c_str());
+			return false;
+		}
+
+		replacement_text = iter->second; 
+
+		num_chars_to_replace += key.length();
+	} else {
+		// for list containers, check if this matches a list modifier
+		Assert(container.is_list());
+
+		if (container.list_data.empty()) {
+			Warning(LOCATION, "Attempt to replace text using empty list container %s", container.container_name.c_str());
+			return false;
+		}
+
+		int modifier_index = 0;
+		for (; modifier_index < MAX_CONTAINER_MODIFIERS; modifier_index++) {
+			if (str_prefix(Container_modifiers[modifier_index].name, text.c_str() + lookHere)) {
+				break;
+			}
+		}
+
+		if (modifier_index == MAX_CONTAINER_MODIFIERS) {
+			Warning(LOCATION,
+				"sexp_replace_container_refs_with_values() found a list container called %s to replace but the modifer "
+				"is not recognised",
+				container.container_name.c_str());
+			return false;
+		}
+
+		int data_index = 0;
+		size_t number_length = 0;
+		SCP_string number_string;
+		auto &list_data = container.list_data;
+		auto list_it = list_data.begin();
+
+		switch (modifier_index) {
+			case SNF_CONTAINER_GET_FIRST:
+				replacement_text = list_data.front();
+				break;
+
+			case SNF_CONTAINER_GET_LAST:
+				replacement_text = list_data.back();
+				break;
+
+			case SNF_CONTAINER_REMOVE_FIRST:
+				replacement_text = list_data.front();
+				list_data.pop_front();
+				break;
+
+			case SNF_CONTAINER_REMOVE_LAST:
+				replacement_text = list_data.back();
+				list_data.pop_back();
+				break;
+
+			case SNF_CONTAINER_GET_RANDOM:
+				data_index = rand_internal(0, (int)list_data.size() -1);
+				replacement_text = *std::next(list_it, data_index);
+				break;
+
+			case SNF_CONTAINER_REMOVE_RANDOM:
+				data_index = rand_internal(0, (int)list_data.size() -1);
+				std::advance(list_it, data_index);
+				replacement_text = *list_it;
+				list_data.erase(list_it);
+				break;
+
+			case SNF_CONTAINER_AT_INDEX:
+				number_string = text.substr(lookHere + strlen(Container_modifiers[SNF_CONTAINER_AT_INDEX].name));
+				Assert(!number_string.empty());
+				data_index = atoi(number_string.c_str());
+				Assert(data_index < (int)list_data.size());
+				replacement_text = *std::next(list_it, data_index);
+
+				// include spaces, in case the FREDder put one between "At" and the index
+				number_length = strspn(number_string.c_str(), " 0123456789");
+				break;
+
+			default:
+				Warning(LOCATION, "sexp_replace_container_refs_with_values() found a container called %s to replace but the modifer is not recognised", container.container_name.c_str());
+				return false;
+		}
+
+		num_chars_to_replace += strlen(Container_modifiers[modifier_index].name);
+
+		// for the "At" modifier, we need to include the chars used by the index
+		if (modifier_index == SNF_CONTAINER_AT_INDEX) {
+			num_chars_to_replace += number_length;
+		}
+
+		// TODO: verify that the char we're about to skip below is a '&'
```

Comment:
How important is it to check that the char we skip is the final `&`?

And what should we do if it isn't an `&` (i.e., the replacement is malformed)?